lib/Segments/SegmentSaveController.php

```diff
@@ -25,6 +33,33 @@ public function save(array $data = []): SegmentEntity {
     return $this->segmentsRepository->createOrUpdate($name, $description, $id);
   }
 
+  public function duplicate(SegmentEntity $segmentEntity): SegmentEntity {
+    $duplicate = clone $segmentEntity;
+    $duplicate->setName(sprintf(__('Copy of %s', 'mailpoet'), $segmentEntity->getName()));
+
+    $this->checkSegmenUniqueName($duplicate->getName(), $duplicate->getId());
+
+    $this->segmentsRepository->persist($duplicate);
+    $this->segmentsRepository->flush();
+    
+    $subscriberSegments = $this->subscriberSegmentRepository->findBy(['segment' => $segmentEntity]);
+    foreach ($subscriberSegments as $subscriberSegment) {
+      $subscriber = $subscriberSegment->getSubscriber();
```

Comment:
Again I am not sure if this is possible in Doctrine, but this makes a separate query for each subscriber, right? Can we prefetch all the subscribers first so that we save the database? I am not sure if that is a good idea.

Reply:
Good point. We can do it. It should be possible by query instead of working with entities. When we use entities, there will be one query per entity, but it will be only one if we use a query.

Reply:
Can't we use some eager loading or something? I am not sure how that works though.